task-management/src/main/webapp/vue-app/tasks-management/components/tasks/TasksList.vue

```diff
@@ -17,6 +18,27 @@
         default: () => [],
       }
     },
+    data() {
+      return {
+        filterTaskCompletedActive: false,
+      }
+    },
+    methods: {
+      updateTaskCompleted(e) {
+        this.filterCompletedActive();
+        if (!this.filterTaskCompletedActive) {
+          window.setTimeout(() => this.tasks = this.tasks.filter((t) => t.task.id !== e.id), 500);
+        }
+      },
+      filterCompletedActive() {
+        this.tasks.forEach(task_elm => {
+          while (task_elm.completed === true) {
+            this.filterTaskCompletedActive = true;
+            break;
+          }
+        })
+      },
+    }
```

Comment:
Combine both functions and use the filter to remove all completed tasks based on their property

Reply:
when the filtre marked as completed active, we shouldn't  remove the task from the list when we clique in the button marked as completed, for that I add  this method filterCompletedActive to test if the filtre active or not